test/properties/electronic/integrals/test_two_body_electronic_integrals.py

```diff
@@ -0,0 +1,455 @@
+            assert np.allclose(ints_mo._matrices[1], 16 * mat_ba)
+            assert np.allclose(ints_mo._matrices[2], 16 * mat_bb)
+            assert np.allclose(ints_mo._matrices[3], 16 * mat_ba.T)
+
+        with self.subTest("Beta custom coeff with only alpha"):
+            transform_beta = ElectronicBasisTransform(
+                ElectronicBasis.AO, ElectronicBasis.MO, 2 * np.eye(2), 3 * np.eye(2)
+            )
+            ints_ao = TwoBodyElectronicIntegrals(ElectronicBasis.AO, (mat_aa, None, None, None))
+            ints_mo = ints_ao.transform_basis(transform_beta)
+            assert np.allclose(ints_mo._matrices[0], 16 * mat_aa)
+            assert ints_mo._matrices[1] is None
+            assert ints_mo._matrices[2] is None
+            assert ints_mo._matrices[3] is None
+
+        with self.subTest("Beta custom coeff"):
+            transform_beta = ElectronicBasisTransform(
+                ElectronicBasis.AO, ElectronicBasis.MO, 2 * np.eye(2), 3 * np.eye(2)
+            )
+            ints_ao = TwoBodyElectronicIntegrals(
+                ElectronicBasis.AO, (mat_aa, mat_ba, mat_bb, mat_ba.T)
+            )
+            ints_mo = ints_ao.transform_basis(transform_beta)
+            assert np.allclose(ints_mo._matrices[0], 16 * mat_aa)
+            assert np.allclose(ints_mo._matrices[1], 36 * mat_ba)
+            assert np.allclose(ints_mo._matrices[2], 81 * mat_bb)
+            assert np.allclose(ints_mo._matrices[3], 36 * mat_ba.T)
+
+        with self.subTest("Final basis match"):
+            ints_ao = TwoBodyElectronicIntegrals(ElectronicBasis.MO, (mat_aa, None, None, None))
+            ints_mo = ints_ao.transform_basis(transform)
+            assert ints_ao == ints_mo
+
+        with self.subTest("Inital basis mismatch"):
+            with self.assertRaises(QiskitNatureError):
+                ints_ao = TwoBodyElectronicIntegrals(ElectronicBasis.SO, mat_aa)
+                ints_ao.transform_basis(transform)
+
+    def test_to_spin(self):
+        """Test to_spin"""
+        mat_aa = np.arange(16).reshape((2, 2, 2, 2))
+        mat_ba = np.arange(16, 32).reshape((2, 2, 2, 2))
+        mat_bb = np.arange(-16, 0).reshape((2, 2, 2, 2))
+
+        with self.subTest("Only alpha"):
+            ints = TwoBodyElectronicIntegrals(ElectronicBasis.MO, (mat_aa, None, None, None))
+            mat_so = ints.to_spin()
+            expected = (
+                [
+                    [
+                        [
+                            [0.0, -1.0, 0.0, 0.0],
+                            [-4.0, -5.0, 0.0, 0.0],
+                            [0.0, 0.0, 0.0, 0.0],
+                            [0.0, 0.0, 0.0, 0.0],
+                        ],
+                        [
+                            [-2.0, -3.0, 0.0, 0.0],
+                            [-6.0, -7.0, 0.0, 0.0],
+                            [0.0, 0.0, 0.0, 0.0],
+                            [0.0, 0.0, 0.0, 0.0],
+                        ],
+                        [
```

Comment:
Shall we store these big matrices in separate files?

Reply:
Yes, hence the `TODO` label in these places. I did not have the time to do this before leaving.

Reply:
Ok, I only noticed the label in the last file.